src/Request.js

```diff
@@ -50,6 +50,18 @@ export default class Request {
             .join('&');
     }
 
+    static format_payload (payload, content_type) {
+        if (
+            typeof payload !== 'object'
+            || content_type === 'application/json'
```

Comment:
Why do we need to change from the existing behaviour of

```
(!this.headers['Content-Type'] && this.data)
```
??

Was there another bug that we needed to fix?

If so, please identify clearly the case (and write a unit test for it)

Reply:
Yes, I'm changing this condition to fix the bug we described on the ticket.

If content type is set, this condition wrongfully assumes that the data has to be serialized using `JSON.stringify`. We don't want to determine if we will use `JSON.stringify` or `Request.stringify` only on the existence of `Content-Type` header alone. With this change, the determination is now based on what the value of `Content-Type` actually is.

The unit tests is already written and committed: https://github.com/anyTV/cuddle/blob/4792cb0e7c014883e4b7beac25dad605920c8608/test/functional/retry.js#L85-L137

Reply:
I don't think this change is necessary with the addition of _formatted_payload? Since with the check, we already won't call format_payload on retry.

The test case you quoted doesn't cover this case -> without retry

```
auto_format = true
content_type = null
payload = "test"

// old code
payload  = Request.stringify(this.data)

// your code
payload = this._formatted_payload = JSON.stringify(payload);
```